src/einsteinpy/coordinates/velocity.py

```diff
@@ -0,0 +1,166 @@
+import astropy.units as u
+import numpy as np
+
+from einsteinpy.coordinates import BoyerLindquist, Cartesian, Spherical
+
+
+class CartesianDifferential(Cartesian):
+    """
+    Class for calculating and transforming the velocity
+    """
+
+    @u.quantity_input(
+        x=u.km, y=u.km, z=u.km, v_x=u.km / u.s, v_y=u.km / u.s, v_z=u.km / u.s
+    )
+    def __init__(self, x, y, z, v_x, v_y, v_z):
+        super(CartesianDifferential, self).__init__(x, y, z)
+        self.v_x = v_x
+        self.v_y = v_y
+        self.v_z = v_z
+
+    def spherical_differential(self):
+        """
+        Function to convert cartesian to spherical coordinates
+        """
+        transformed_spherical = self.to_spherical()
+        n1 = self.x ** 2 + self.y ** 2
+        n2 = n1 + self.z ** 2
+        v_r = (self.x * self.v_x + self.y * self.v_y + self.z * self.v_z) / np.sqrt(n2)
+
+        v_t = (
+            (self.z * (self.x * self.v_x + self.y * self.v_y) - n1 * self.v_z)
+            / (n2 * np.sqrt(n1))
+            * u.rad
+        )
+
+        v_p = (-1 * (self.v_x * self.y - self.x * self.v_y) / n1) * u.rad
+        return SphericalDifferential(
+            transformed_spherical.r,
+            transformed_spherical.theta,
+            transformed_spherical.phi,
+            v_r,
+            v_t,
+            v_p,
+        )
+
+    def boyerlindquist_differential(self, a):
```

Comment:
Rename this to `bl_differential`

Reply:
Also, a should be with units :)

Reply:
https://github.com/einsteinpy/einsteinpy/pull/176/files#diff-517ed4ee5def3bce49f42958bec83fc4R207
See this